aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -118,15 +118,52 @@ impl<'e, E: ExecutorView> StorageAdapter<'e, E> {
             };
 
             let buf_size = resource_size(&buf);
-            Ok((buf, buf_size + group_size as usize))
-        } else {
-            let state_key = resource_state_key(address, struct_tag)?;
-            let buf = self
-                .executor_view
-                .get_resource_bytes(&state_key, maybe_layout)?;
-            let buf_size = resource_size(&buf);
-            Ok((buf, buf_size))
+            return Ok(Some((buf, buf_size + group_size as usize)));
         }
+        Ok(None)
+    }
+
+    fn get_any_resource_with_layout(
+        &self,
+        address: &AccountAddress,
+        struct_tag: &StructTag,
+        metadata: &[Metadata],
+        maybe_layout: Option<&MoveTypeLayout>,
+    ) -> PartialVMResult<(Option<Bytes>, usize)> {
+        if let Some(result) =
+            self.get_resource_group_member_bytes(address, struct_tag, metadata, maybe_layout)?
+        {
+            return Ok(result);
+        }
+
+        let state_key = resource_state_key(address, struct_tag)?;
+        let buf = self
+            .executor_view
+            .get_resource_bytes(&state_key, maybe_layout)?;
+        let buf_size = resource_size(&buf);
+        Ok((buf, buf_size))
+    }
+
+    fn get_any_resource_size_with_layout(
+        &self,
+        address: &AccountAddress,
+        struct_tag: &StructTag,
+        metadata: &[Metadata],
+        maybe_layout: Option<&MoveTypeLayout>,
+    ) -> PartialVMResult<ResourceSizeInfo> {
+        if let Some((bytes, bytes_loaded)) =
+            self.get_resource_group_member_bytes(address, struct_tag, metadata, maybe_layout)?
```

Comment:
we also have APIs for getting size of a resource group member, but we can add a TODO and connect as a follow-up?

We could even consider landing this without connecting here - just using the default implementation, and then having the connection PR separate (which can also connect resource groups). It may not be super important here just sometimes a good practice, @georgemitenkov what do you think?